classes/migration/upgrade/v3_4_0/I9627_AddUsageStatsTemporaryTablesIndexes.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I9627_AddUsageStatsTemporaryTablesIndexes.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I9627_AddUsageStatsTemporaryTablesIndexes
+ *
+ * @brief Add an index to temporary usage stats tables to fix/improve the removeDoubleClicks and compileUniqueClicks query.
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+
+class I9627_AddUsageStatsTemporaryTablesIndexes extends Migration
+{
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        $sm = Schema::getConnection()->getDoctrineSchemaManager();
+        Schema::table('usage_stats_total_temporary_records', function (Blueprint $table) use ($sm) {
+            $indexesFound = $sm->listTableIndexes('usage_stats_total_temporary_records');
+            if (!array_key_exists('ust_load_id_context_id_ip', $indexesFound)) {
```

Comment:
I think you've added this to cover the users that already added the indexes, right? I guess the users won't be using the same index name, then it's not going to work.

In case the index name is the same, and an error happens, it's also not our fault (given the user isn't supposed to change the schema).

Reply:
I added it because I would ask the two affected users to immediately patch the installation with the PRs that I add to this issue...

Reply:
Just tagging @asmecher to check if he agree and possibly look out for this on next reviews (ensure migrations can be re-executed).